console_backend/src/tabs/advanced_tab/advanced_imu_tab.rs

```diff
@@ -97,18 +98,25 @@ impl AdvancedImuTab {
     /// # Parameters
     /// - `msg`: MsgImuAux to extract data from.
     pub fn handle_imu_aux(&mut self, msg: MsgImuAux) {
-        match msg.imu_type {
-            0 => {
-                self.imu_temp = 23_f64 + msg.temp as f64 / 512_f64;
+        match msg.imu_type() {
+            Ok(imu) => {
+                match imu {
+                    ImuType::BoschBmi160 => {
+                        self.imu_temp = 23_f64 + msg.temp as f64 / 512_f64;
+                    }
+                    ImuType::StMicroelectronicsAsm330Llh => {
+                        self.imu_temp = 25_f64 + msg.temp as f64 / 256_f64;
+                    }
+                    ImuType::MurataScha634D03 => {
+                        self.imu_temp = 25_f64 + msg.temp as f64 / 256_f64;
+                    }
+                    ImuType::TdkIam20680Hp => {
+                        self.imu_temp = 25_f64 + msg.temp as f64 / 256_f64;
```

Comment:
same as above^

Reply:
```((TEMP_OUT – RoomTemp_Offset)/Temp_Sensitivity) + 25degC```
seems per  https://invensense.tdk.com/wp-content/uploads/2021/11/DS-000409-IAM-20680HP-v1.2-Typ.pdf
needs multiple params?

Reply:
Looks like those are defined here (section 3.3.2):

![image](https://user-images.githubusercontent.com/183436/228680735-85f91c05-c2bf-4d0c-b04f-5bde518ee907.png)

Reply:
hmmm yea was thinkin but number doesnt seem that nice compared to the others... guess theres no other choice 😢